wasm/Cargo.toml

```diff
@@ -42,7 +42,7 @@ license.workspace = true
 debug = []
 
 [dependencies]
-iroha_data_model = { version = "=2.0.0-pre-rc.13", path = "../data_model", default-features = false }
+iroha_data_model = { version = "=2.0.0-pre-rc.13", path = "../data_model", default-features = false, features = ["transparent_api"]}
```

Comment:
this is not ok

Reply:
`iroha_wasm` is not an internal crate. When a user goes to write a smart contract they are pulling it into their project directly or transitively. Internal crate is one that does not end up in a dependency graph of a user written application or smart contract. 

You can handle this by:
1. providing getters
2. providing a method for destructuring like `impl From<Transaction> for (.., .., ..)`

We can revisit this after we've settled other discussions

Reply:
> Okay, what are the cons of using it like so?

with dynamic linking all data model structs are going to be opaque

Reply:
Updated. Check re-review please